controllers/profile.js

```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
             })
             .populate({
                 path: "workPlace",
+            })
+            .populate({
+                path: "workTime",
+                populate: {
+                    path: "patientData",
+                    populate: "userData",
+                },
             });
-        const userData = {
-            name: doctor.userData.name,
-            lastName: doctor.userData.lastName,
-            photo: doctor.userData.photo,
-            phone: doctor.userData.phone,
-            email: doctor.userData.email,
-            experience: doctor.experience,
-            direction: doctor.direction,
-            workPlace: {
-                name: doctor.workPlace.name,
-                address: doctor.workPlace.address,
-            },
-            workTime: doctor.workTime,
-            patients: doctor.patients,
-        };
-        res.status(200).json(userData);
+        if (doctor) {
+            const workTime = doctor.workTime.map((time) => ({
+                doctorData: time.doctorData,
+                date: time.date,
+                time: time.time,
+                patientName: time.patientData?.userData.name,
+                patientLastName: time.patientData?.userData.lastName,
+                patientPhoto: time.patientData?.userData.photo,
+                idWorkTime: time._id,
+            }));
+            const doctorData = {
+                name: doctor.userData.name,
+                lastName: doctor.userData.lastName,
+                photo: doctor.userData.photo,
+                phone: doctor.userData.phone,
+                email: doctor.userData.email,
+                experience: doctor.experience,
+                direction: doctor.direction,
+                workPlace: {
+                    name: doctor.workPlace.name,
+                    address: doctor.workPlace.address,
+                },
+                workTime: workTime,
+                patients: doctor.patients,
+            };
+            res.status(200).json(doctorData);
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: "Server error" });
```

Comment:
message is repeated with the same string, makes sense to make it a reusable constant